stackslib/src/chainstate/burn/db/v2/test_helpers.rs

```diff
@@ -0,0 +1,1011 @@
+use std::ops::Deref;
+
+use crate::chainstate::stacks::index::trie_db::TrieDb;
+
+use super::{SortitionDb, SortitionDbTransaction};
+
+pub struct InMemorySortitionDb {
+
+}
+
+impl SortitionDbTransaction for InMemorySortitionDb {
```

Comment:
Wow, this trait is _huge_!  This, to me, suggests that this is the wrong place to add a layer of indirection -- it looks like the trait is really just coding to a particular implementation.  Also, a lot of these methods are specific to Stacks 2.x, and are gone in Nakamoto.  Perhaps there's a way to decompose this trait into a few smaller, well-defined traits?

Reply:
Haha yes :) I definitely think that it could be broken up (like I did for `ClarityDb*` - but since I'm less proficient in these areas I opted to simply take them "as-is" from their struct impl blocks.

Reply:
An afterthought is that we should _probably_ keep it as a big trait _to begin with_, and split it up in a later iteration - just to try to minimize the changes in a single PR as much as possible.